src/org/sosy_lab/common/configuration/ConfigurationBuilder.java

```diff
@@ -260,9 +262,26 @@ public ConfigurationBuilder loadFromResource(Class<?> contextClass, String resou
     @Var Path sourcePath;
     @Var String sourceString;
     try {
+      // try to access the requested file.
```

Comment:
This does not try to access the given file, so please rewrite the comment.

Given that the whole code is quite non-obvious as it relies on the fact that `FileSystemProvider.newFileSystem` has some side effect, this really needs good documentation, with an explanation of the problem that this solves, why it solves it, and a link to the source of the JAR.

Reply:
I restructured the code and added some documentation.
Please ask if anything remains unclear.

Reply:
Well, this comment is still as wrong as before, right?